File: dotnet-core/Tea/Decorators/Drivers/LoggingBrowser.cs
using System.Collections.Generic;
using OpenQA.Selenium;
using System;

namespace Tea
{
    public class LoggingBrowser : BrowserDecorator
    {
        public LoggingBrowser(WebBrowser Browser) : base(Browser){}

        public override IWebDriver GetDriver()
        {
            return base.GetDriver();
        }
        
        public override Element FindElement(By locator)
        {
            Console.WriteLine($"Find Element: " + locator.ToString());
            return Browser?.FindElement(locator);
        }

        public override List<Element> FindElements(By locator)
        {
            Console.WriteLine($"Find Elements: " + locator.ToString());
            return Browser?.FindElements(locator);
        }

        public override void GoTo(string url)
        {
            Console.WriteLine($"Go to URL: {url}");
            Browser?.GoTo(url);
        }

        public override void Quit()
        {
            Console.WriteLine($"Closing browser.");
            Browser?.Quit();
        }

        public override void Start(BrowserList browser)
        {
            Console.WriteLine($"Starting browser: {(Enum.GetName(typeof(BrowserList), browser))}");
            Browser?.Start(browser);
        }
    }
}

File: java/tea-framework/src/main/java/com/tea/framework/SeleniumWrapper/Drivers/LoggingBrowser.java
package com.tea.framework.SeleniumWrapper.Drivers;

import com.tea.framework.SeleniumWrapper.Elements.Element;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

import java.util.List;

import static com.tea.framework.SeleniumWrapper.Elements.LogElement.write;

public class LoggingBrowser extends BrowserDecorator {
    public LoggingBrowser(WebBrowser Browser) {
        super(Browser);
    }

    @Override
    public WebDriver GetDriver() {
        return this.browser.GetDriver();
    }

    @Override
    public Element FindElement(By locator) {
        write("Find Element: " + locator.toString());
        return browser.FindElement(locator);
    }

    @Override
    public List<Element> FindElements(By locator) {
        write("Find Elements: " + locator.toString());
        return browser.FindElements(locator);
    }

    @Override
    public void GoTo(String url) {
        write("Go to URL: {url}");
        browser.GoTo(url);
    }

    @Override
    public void Quit() {
        write("Closing browser.");
        browser.Quit();
    }

    @Override
    public void Start(BrowserList browser) {
        write("Starting browser: {(Enum.GetName(typeof(BrowserList), browser))}");
        this.browser.Start(browser);
    }
}